lib/abr/simple_abr_manager.js

```diff
@@ -122,12 +126,14 @@ shaka.abr.SimpleAbrManager = class {
 
     const enumerate = (it) => shaka.util.Iterables.enumerate(it);
     for (const {item, next} of enumerate(sortedVariants)) {
+      const itemBandwidth = this.getPlaybackRate() * item.bandwidth;
```

Comment:
This part of the code is just a VV log. It needed to be changed, but it's not the .
You should add a playbackRate parameter or something to `shaka.util.StreamUtils.meetsRestrictions` (see the call inside `shaka.abr.SimpleAbrManager.filterAndSortVariants_`); that's where the ABR manager actually checks the bandwidths of the variants.
Similarly, you will want to modify `shaka.util.StreamUtils.applyRestrictions`, since it also calls `shaka.util.StreamUtils.meetsRestrictions`. This will make the "soft restrictions" also take playbackRate into account.
Finally, `shaka.util.StreamUtils.meetsRestrictions` is also called by `shaka.util.ManifestFilter.filterByRestrictions`. That method is only called by the storage engine, though, and playbackRate doesn't really matter in that case, so you can probably just pretend that playbackRate is always 1 for that.

Reply:
@theodab, I think you might be mistaken.  The goal isn't to apply playbackRate to restrictions, but to track selection.  And this section is where that happens.  It's more than just a v2 log point.  You may need to expand the code a bit to see lines 140-143 where minBandwidth and maxBandwidth are used.

Reply:
Ah, I see.